### PR TITLE
Document changes to Flux string interpolation

### LINKED ISSUE #2314: Update string interpolation
**PR:** https://github.com/influxdata/flux/pull/3556

Updates to string interpolation behavior. Previously, the string interpolated segment had to be a string. Now, any expression that resolves to a basic type (String, Int, Uint, Bool, Time, Duration) can be used in the string interpolated segment. 

https://docs.influxdata.com/influxdb/v2.0/reference/flux/language/string-interpolation/

Contact @Bolladeen with questions/review.

### PR DESCRIPTION
**Pull request by pierwill**
Closes #2314

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/reference/flux/language/string-interpolation.md [hunk 1]
```diff
@@ -27,19 +27,14 @@ name = "John"
...
 
-{{% note %}}
-#### Flux only interpolates string values
-Flux currently interpolates only string values ([IMP#1775](https://github.com/influxdata/flux/issues/1775)).
-Use the [string() function](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/type-conversions/string/)
-to convert non-string values to strings.
-
-```js
-count = 12
+String interpolation expressions must satisfy the Stringable constraint.
```
Comment (sanderson) [inline content/influxdb/v2.0/reference/flux/language/string-interpolation.md:30]: - Add the "Stringable" constraint to this doc: /influxdb/v2.0/reference/flux/language/types/#type-constraints
- Link "Stringable constraint" to the new section

File: content/influxdb/v2.0/reference/flux/language/string-interpolation.md [hunk 2]
```diff
@@ -27,19 +27,14 @@ name = "John"
...
 
-"I currently have ${string(v: count)} cats."
-```
-{{% /note %}}
+Arbitrary Expression Interpolation example:
 
+    n = duration(v: "1m") 
+    "the answer is ${n}" // the answer is 1m
+    t0 = time(v: "2016-06-13T17:43:50.1004002Z")
+    "the answer is ${t0}" // the answer is 2016-06-13T17:43:50.1004002Z
```
Comment (sanderson) [inline content/influxdb/v2.0/reference/flux/language/string-interpolation.md:37]:
~~~suggestion
##### Arbitrary expression interpolation example:
```js
n = duration(v: "1m") 
"the answer is ${n}"
// the answer is 1m

t0 = time(v: "2016-06-13T17:43:50.1004002Z")
"the answer is ${t0}"
// the answer is 2016-06-13T17:43:50.1004002Z
```
~~~

**Review by sanderson — CHANGES_REQUESTED** [review]
A few things:

- This change is bound to a Flux release that hasn't happened yet. This shouldn't be merged into `master`. We need to create a `flux-0.109` branch and use that as the base branch for these changes.
- These changes are only part of a much broader set of changes that need to take place:
    - Updating this doc: https://docs.influxdata.com/influxdb/v2.0/reference/flux/language/string-interpolation/
    - Updating all the string interpolation examples through the docs that use the string function to convert interpolated non-string values to strings: `"${string(v: ... )}"`.

  The challenge here is that this version of Flux will not be immediately available in each version of InfluxDB. It will roll out to Cloud, and hopefully be part of InfluxDB 2.0.5. Not sure exactly how to handle this. My first thought is to leave the `string()` examples in there since they're still valid and change them once the new version of Flux is widely distributed.

**Comment by pierwill** [comment]
Thanks, @sanderson. I've created the `flux-0.109` branch.

**Review by sanderson — APPROVED** [review]

**Comment by sanderson** [comment]
Merging this now without updating the string interpolation examples. There's a defect in the current string interpolation implementation that will be fixed in a future release. Once it's released, we'll go through and update all the examples.

**Pull request merged**